### ISSUE TITLE
Why is SAF loss negative，FreeMatch

### ISSUE THREAD

**Issue by yumudry** [microsoft/Semi-supervised-learning#231]
# calculate entropy loss
            if mask.sum() > 0:
               ent_loss, _ = entropy_loss(mask, logits_x_ulb_s, self.p_model, self.label_hist)
            else:
               ent_loss = 0.0
            # ent_loss = 0.0
            total_loss = sup_loss + self.lambda_u * unsup_loss + self.lambda_e * ent_loss

        out_dict = self.process_out_dict(loss=total_loss, feat=feat_dict)
        log_dict = self.process_log_dict(sup_loss=sup_loss.item(), 
                                         unsup_loss=unsup_loss.item(), 
                                         ent_loss=ent_loss.item(),
                                         total_loss=total_loss.item(), 
                                         util_ratio=mask.float().mean().item())
       ent_loss=ent_loss.item()             
      
Ent_ross, I printed this loss value and it's all negative                  
![image](https://github.com/user-attachments/assets/163f3ec6-178a-4ec4-b9b4-114d488e7aaa)
